src/js/utils.js

```diff
@@ -468,23 +469,26 @@ function firstPartyProtectionsEnabled(domain, badger) {
         firstParties.push(extractedUrls);
       }
     }
-    // clean up the gathered sets of url matches into a single flat list
-    badger.firstPartiesList = [].concat.apply([], firstParties);
+    return [].concat.apply([], firstParties);
   }
-  // check for presence of given domain in firstparties list
-  for (let url_pattern of badger.firstPartiesList) {
-    // account for wildcards in entries on firstPartiesList & avoid false positives
-    if (url_pattern.startsWith("*")) {
-      // compare against pattern without '*.' and the domain without 'www.'
-      if (url_pattern.slice(2) == domain.slice(4)) {
+
+  return function (tab_host) {
+    if (!firstPartiesList) {
+      firstPartiesList = getFirstParties();
+    }
+
+    for (let url_pattern of firstPartiesList) {
+      if (url_pattern.startsWith("*")) {
+        if ((url_pattern.slice(2) == tab_host.slice(4)) || url_pattern.slice(2) == tab_host.slice(2)) {
```

Comment:
You shouldn't have to assume anything about what tab host subdomains look like. Does the tab host end with the wildcard pattern minus the wildcard?